external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpout.java

```diff
@@ -395,7 +387,7 @@ private boolean emitOrRetryTuple(ConsumerRecord<K, V> record) {
         } else if (emitted.contains(msgId)) {   // has been emitted and it is pending ack or fail
             LOG.trace("Tuple for record [{}] has already been emitted. Skipping", record);
         } else {
-            if (kafkaConsumer.committed(tp) != null && (kafkaConsumer.committed(tp).offset() >= kafkaConsumer.position(tp))) {
+            if (kafkaConsumer.committed(tp) != null && (kafkaConsumer.committed(tp).offset() > kafkaConsumer.position(tp))) {
```

Comment:
Is this change to address https://issues.apache.org/jira/browse/STORM-2844 ?

Reply:
No, it's just to get a test to pass that happened to hit this case. 2844 is still an issue.